File: client/types/product.ts
import { IClassification, ISiteDetail } from 'types'

export interface IProductImage {
    uuid: string
    name: string
    apiPath: string
}
export interface IProduct {
    uuid: string
    name: string
    description: string
    price: number
    category: IClassification
    tags: Array<IClassification>
    productImages: Array<IProductImage>
    siteDetails: Array<ISiteDetail>
    isActive: boolean
}

export interface IGetProductsParams {
    mode: 'all' | 'active'
}

export interface ICarouselItem {
    product: IProduct
    apiPath: string
}

File: backend/internal/interface/http/response/product.go
package response

type ProductClassificationResponse struct {
	UUID string `json:"uuid"`
	Name string `json:"name"`
}

type ProductSalesSiteResponse struct {
	UUID string `json:"uuid"`
	Name string `json:"name"`
}

type ProductSiteDetailResponse struct {
	UUID      string                   `json:"uuid"`
	DetailURL string                   `json:"detailUrl"`
	SalesSite ProductSalesSiteResponse `json:"salesSite"`
}

type ProductImageResponse struct {
	UUID         string `json:"uuid"`
	Name         string `json:"name"`
	DisplayOrder int    `json:"displayOrder"`
	APIPath      string `json:"apiPath"`
}

type ProductResponse struct {
	UUID          string                          `json:"uuid"`
	Name          string                          `json:"name"`
	Description   string                          `json:"description"`
	Price         int                             `json:"price"`
	IsRecommend   bool                            `json:"isRecommend"`
	IsActive      bool                            `json:"isActive"`
	Category      ProductClassificationResponse   `json:"category"`
	Target        ProductClassificationResponse   `json:"target"`
	Tags          []ProductClassificationResponse `json:"tags"`
	ProductImages []ProductImageResponse          `json:"productImages"`
	SiteDetails   []ProductSiteDetailResponse     `json:"siteDetails"`
}

type CreateProductResponse struct {
	UUID string `json:"uuid"`
}

type CategoryProductsResponse struct {
	Category ProductClassificationResponse `json:"category"`
	PageInfo CursorPageInfoResponse        `json:"pageInfo"`
	Products []*ProductResponse            `json:"products"`
}

type CarouselItemResponse struct {
	Product *ProductResponse `json:"product"`
	APIPath string           `json:"apiPath"`
}
